homeassistant/components/technove/coordinator.py

```diff
@@ -0,0 +1,46 @@
+"""DataUpdateCoordinator for TechnoVE."""
+from __future__ import annotations
+
+from technove import Station as TechnoVEStation, TechnoVE, TechnoVEError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import CALLBACK_TYPE, HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER, SCAN_INTERVAL
+
+
+class TechnoVEDataUpdateCoordinator(DataUpdateCoordinator[TechnoVEStation]):
+    """Class to manage fetching TechnoVE data from single endpoint."""
+
+    config_entry: ConfigEntry
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        *,
+        entry: ConfigEntry,
+    ) -> None:
+        """Initialize global TechnoVE data updater."""
+        self.technove = TechnoVE(
+            entry.data[CONF_IP_ADDRESS], session=async_get_clientsession(hass)
+        )
+        self.unsub: CALLBACK_TYPE | None = None
+
+        super().__init__(
+            hass,
```

Comment:
If you move this to the top of the constructor. The super constructor will set self.config_entry. you have to retype it tho, check out the YouTube coordinator for an example

Reply:
Done, I moved it to the top of the constructor. Just for reference, I based my work on the WLED integration, which calls the super constructor last.